src/main/java/com/amazon/jenkins/ec2fleet/FleetStateStats.java

```diff
@@ -95,10 +110,24 @@ public static FleetStateStats readClusterState(final AmazonEC2 ec2, final String
             throw new IllegalStateException("Fleet " + fleetId + " can't be described");
 
         final SpotFleetRequestConfig fleetConfig = fleet.getSpotFleetRequestConfigs().get(0);
+        final SpotFleetRequestConfigData fleetRequestConfig = fleetConfig.getSpotFleetRequestConfig();
+
+        // Index configured instance types by weight:
+        final Map<String, Double> instanceTypeWeight = new HashMap<>();
+        for (SpotFleetLaunchSpecification launchSpecification : fleetRequestConfig.getLaunchSpecifications()) {
+            final String instanceType = launchSpecification.getInstanceType();
+            final Double instanceWeight = launchSpecification.getWeightedCapacity();
+            final Double existingWeight = instanceTypeWeight.get(instanceType);
+            if (instanceWeight == null  || (existingWeight != null && existingWeight > instanceWeight)) {
```

Comment:
I think you can skip this logic, you cannot have diff weights for same instance type